irods_testing_environment/test_runner.py

```diff
@@ -85,16 +87,19 @@ def result_string(self):
         r = '-----\nresults for [{}]\n'.format(self.name())
 
         r = r + '\tpassed tests:\n'
-        for test, duration in self.passed_tests().items():
-            r = r + '\t\t[[{:>9.4f}]s]\t[{}]\n'.format(duration, test)
+        for test, duration in self.passed_tests():
+            # TODO: a test list of type None may not indicate that all tests ran
+            r = r + '\t\t[[{:>9.4f}]s]\t[{}]\n'.format(duration, test or 'all tests')
```

Comment:
What causes `'all tests'` to be printed?

Reply:
`test or 'all tests'` is equivalent to `test if test else 'all tests'` where `test == None` evaluates to `False`.

Reply:
It was about L92. I was trying to understand what triggers the `'all tests'` text to be printed.

Additional comments are always good too :)

Reply:
Okay, I will add a blurb to each test runner to describe the None case.

Reply:
Done